src/PrestaShopBundle/Controller/Admin/Improve/Shipping/PreferencesController.php

```diff
@@ -42,7 +41,7 @@ class PreferencesController extends FrameworkBundleAdminController
      *
      * @param Request $request
      *
-     * @Template("@PrestaShop/Admin/Improve/Shipping/Preferences/preferences.html.twig")
+     * @AdminSecurity("is_granted(['read', 'update', 'create', 'delete'], request.get('_legacy_controller'))")
```

Comment:
Isn't more logical to only ask for READ, especially if the test is: read OR update OR create OR delete?

Reply:
Bigger issue is https://github.com/PrestaShop/PrestaShop/issues/12109
But the short answer: on the BO, I think if you have been given the right to create something, update it or delete it, then you should be able to see it displayed.

If everybody was very rigorous about how they grant accesses, we could restrict it to "read" only. I'm handling the "write but no read" rule here 😅

Reply:
(Wrong button chosen)

I've been looking in other controller, and for instance in BackupController & EmployeeController, only the "read" permission is requested, which makes sense in my opinion.

What do you think about updating this line?